src/tools/rbd_mirror/Mirror.cc

```diff
@@ -30,15 +30,25 @@ namespace mirror {
 Mirror::Mirror(CephContext *cct) :
   m_cct(cct),
   m_lock("rbd::mirror::Mirror"),
-  m_local(new librados::Rados())
+  m_local(new librados::Rados()),
+  m_mirror_thread(this)
 {
   cct->lookup_or_create_singleton_object<Threads>(m_threads,
                                                   "rbd_mirror::threads");
 }
 
-void Mirror::handle_signal(int signum)
+Mirror::~Mirror()
 {
+  dout(20) << "terminating" << dendl;
+
   m_stopping.set(1);
+  {
+    Mutex::Locker l(m_lock);
+    m_cond.Signal();
```

Comment:
Wouldn't just signaling the condition be enough of a change to interrupt the 30 second sleep?